File: LC_Gen_Codes/2900_no_setting.py
class Solution:
    def getWordsInLongestSubsequence(self, n: int, words: List[str], groups: List[int]) -> List[str]:
        subsequence = []
        prev_group = None
        
        for i in range(n):
            if prev_group is None or groups[i] != prev_group:
                subsequence.append(words[i])
                prev_group = groups[i]
        
        return subsequence

File: LC_Gen_Codes_CPlusPlus/2900_no_setting.cpp
import java.util.ArrayList;
import java.util.List;

class Solution {
    public List<String> getWordsInLongestSubsequence(int n, List<String> words, List<Integer> groups) {
        List<String> subsequence = new ArrayList<>();
        Integer prev_group = null;
        
        for (int i = 0; i < n; i++) {
            if (prev_group == null || !groups.get(i).equals(prev_group)) {
                subsequence.add(words.get(i));
                prev_group = groups.get(i);
            }
        }
        
        return subsequence;
    }
}